src/util/formChangeCount.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  ConfigurationRowFormikProps,
+  ConfigurationRowFormikValues,
+} from "components/ConfigurationRow";
+import { FormDevice, FormDeviceValues } from "util/formDevices";
+import { ResourceLimitsFormValues } from "components/forms/ResourceLimitsForm";
+import { InstanceEditDetailsFormValues } from "pages/instances/EditInstance";
+
+const getPrimitiveFieldChanges = (
+  formik: ConfigurationRowFormikProps,
+): number => {
+  let changeCount = 0;
+
+  const ignoredFields = [
+    "readOnly",
+    "profiles",
+    "limits_cpu",
+    "limits_memory",
+    "devices",
+    "barePool",
+  ];
+
+  for (const key in formik.values) {
+    if (ignoredFields.includes(key)) {
+      continue;
+    }
+    const keyType = key as keyof ConfigurationRowFormikValues;
+    if (formik.values[keyType] !== formik.initialValues[keyType]) {
+      changeCount++;
+      console.log(key, formik.values[keyType], formik.initialValues[keyType]);
+    }
+  }
+
+  return changeCount;
+};
+
+const getProfileChanges = (formik: ConfigurationRowFormikProps): number => {
+  if (!Object.hasOwn(formik.values, "profiles")) {
+    return 0;
+  }
+
+  const initProfiles = (formik.initialValues as InstanceEditDetailsFormValues)
+    .profiles;
+  const profiles = (formik.values as InstanceEditDetailsFormValues).profiles;
+
+  const hasOrderChange = () => {
```

Comment:
since profile names are unique, maybe it's easier to iterate the items in `initProfiles` and check if the corresponding item in `profiles` with the same index have the same name? Then do an early exit if any mismatch is found

Reply:
yes, good idea -- changed this to be more simple and accurate.